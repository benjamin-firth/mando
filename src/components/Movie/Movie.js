import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const Movie = ({ title, episode, release, opening, image, filterMovie }) => {
  return (
    <article className="movie">
      <div>
        <h2>{title}</h2>
        <p>Episode #{episode}</p>
        <p>Release Year: {release}</p>
        <NavLink id={episode} className='character-button' to={`/characters-page/${episode}`}>View Characters</NavLink>
      </div>
      <img src={image} className="movie-picture" alt="Movie Poster" />
    </article>
  )
}

Movie.propTypes = {
  title: PropTypes.string,
  episode: PropTypes.number,
  release: PropTypes.string,
  opening: PropTypes.string,
  image: PropTypes.string,
  filterMovie: PropTypes.func
}

export default Movie;
