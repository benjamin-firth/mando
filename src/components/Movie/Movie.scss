.movie {
  background-color: $secondary-background-color;
  border-radius: 5px;
  display: flex;
  height: auto;
  justify-content: space-between;
  margin: auto;
  margin-top: 20px;
  padding-left: 10px;
  width: 80vw;
  div {
    width: 50vw;
    text-align: left;
    .character-button {
      background-color: black;
      color: white;
      display: inline-block;
      font-weight: bold;
      height: 5vh;
      padding: 10px 0 0 0;
      text-align: center;
      text-decoration: none;
      text-justify: center;
      width: 40vw;
    }
    .character-button:hover {
      transform: scale(1.02);
    }
  }
  .movie-picture {
    max-height: 220px;
    width: 30vw;
    object-fit: cover;
    border-left: 2px solid yellow;
  }
}