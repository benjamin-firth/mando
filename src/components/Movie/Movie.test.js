import React from 'react';
import { shallow } from 'enzyme';
import Movie from './Movie.js';

describe('Movie', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Movie />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})