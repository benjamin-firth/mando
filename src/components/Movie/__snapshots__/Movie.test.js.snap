// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match the snapshot 1`] = `
<article
  className="movie"
>
  <div>
    <h2 />
    <p>
      Episode #
    </p>
    <p>
      Release Year: 
    </p>
    <NavLink
      className="character-button"
      to="/characters-page/undefined"
    >
      View Characters
    </NavLink>
  </div>
  <img
    alt="Movie Poster"
    className="movie-picture"
  />
</article>
`;
